final_task/rss_reader/Handler.py

```diff
@@ -0,0 +1,90 @@
+import json
+import logging
+
+import feedparser
+from Entry import Entry
+
+
+class Handler:
+    """class for handling different options: --version, --json, --limit Limit"""
+
+    def __init__(self, source: str, limit: int, version: float):
+        self.source = source
+        self.limit = limit
+        self.version = version
+        self.parsed = feedparser.parse(self.source)
+        logging.info("Handler object created")
+
+    """options of command line:"""
+
+    def option_version(self) -> None:
+        logging.info("function \"option_version\" started")
+
+        print("version ", self.version)
+
+        logging.info("function \"option_version\" finished")
+
+    def option_json(self) -> None:
+        logging.info("function \"option_json\" started")
+
+        for num, entry in enumerate(self.gen_entries()):
+            if num == self.limit:
+                break
+            self.print_to_json(self.convert_to_dict(entry))
+
+        logging.info("function \"option_json\" finished")
+
+    def option_default(self) -> None:
+        logging.info("function \"option_default\" started")
+
+        self.print_feed()
+        for num, entry in enumerate(self.gen_entries()):
+            if num == self.limit:
+                break
+            entry.print_title()
+            entry.print_date()
+            entry.print_link()
+            entry.print_summary()
+            entry.print_links()
+            if num == self.limit - 1:
+                break
```

Comment:
A question:  

Why do you need this here if you already have almost the same check a bit above?

Reply:
I think that I need this condition because in the cicle: 
"for num, entry in enumerate(self.gen_entries()):
a new entry will be created and only after it I'll get False in this condition:
if num == self.limit:
    break 
i.e an additional object will be created.

Reply:
That is true, but why do you need `if` condition at the top then? you will never hit it before the lower one anyway

Reply:
If limit is equal for example 0 we'll leave the cycle and won't print information about an entry. Or limit is equal to 1 then we'll get the information about entry and leave the cycle because this condition:
 if num == self.limit - 1:
                break
will return true. If this condition would not exist, we would get one more object entry because it creates in the cicle:
"for num, entry in enumerate(self.gen_entries())
These 2 conditions are necessary to not to get one more extra object.
And I don't know what is more important in this case: not to get an extra object or more readable code.

Reply:
Now I get it. Then it is better to remove upper check and make it so that `--limit` must be > 0 if it was specified by user.

Upper check will trigger only in case if `limit == 0` and in any other case the cycle will break at the bottom check.